### DISCUSSION TITLE
401 Unhautorized when creating package with maven

### DISCUSSION THREAD

**Discussion by sblantipodi** [community/community#25385] [Actions]
<p>Hi,<br>
I have this error:<br>
<code>"Failed to retrieve remote metadata org.dpsoftware:FastScreenCapture:1.0.1-SNAPSHOT/maven-metadata.xml: Could not transfer metadata org.dpsoftware:FastScreenCapture:1.0.1-SNAPSHOT/maven-metadata.xml from/to github ([https://maven.pkg.github.com/sblantipodi/JavaFastScreenCapture](https://maven.pkg.github.com/sblantipodi/JavaFastScreenCapture)): Authentication failed for [https://maven.pkg.github.com/sblantipodi/JavaFastScreenCapture/org/dpsoftware/FastScreenCapture/1.0.1-SNAPSHOT/maven-metadata.xml](https://maven.pkg.github.com/sblantipodi/JavaFastScreenCapture/org/dpsoftware/FastScreenCapture/1.0.1-SNAPSHOT/maven-metadata.xml) 401 Unauthorized"</code></p>
<p>when I try to create a maven package from ci.</p>
<p>I have this in the pom.xml</p>
<pre><code>\&lt;distributionManagement\&gt;  
 \&lt;repository\&gt;  
 \&lt;id\&gt;github\&lt;/id\&gt;  
 \&lt;name\&gt;GitHub Packages\&lt;/name\&gt;  
 \&lt;url\&gt;https://maven.pkg.github.com/sblantipodi/JavaFastScreenCapture\&lt;/url\&gt;  
 \&lt;/repository\&gt;  
\&lt;/distributionManagement\&gt;
</code></pre>
<p>but then in CI when I try to do this:</p>
<pre><code>-name: Publish package
run: mvn -B deploy
env:
GITHUB\_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
<p>I got the error.</p>
<p>can you help me please?</p>

**Comment by sblantipodi** [comment]
<p>I succeded in uploading the first 1.0.0-SNAPSHOT the first time, but now it don’t want to get other packages. No care what version I use.</p>
<p>Is this a bug?</p>

**Comment by sblantipodi** [comment]
<p>this guide does not work neither</p>
<p><a href="https://help.github.com/en/actions/language-and-framework-guides/publishing-java-packages-with-maven" rel="nofollow noopener">https://help.github.com/en/actions/language-and-framework-guides/publishing-java-packages-with-maven</a></p>

**Comment by sblantipodi** [comment, marked as answer]
<p>I’m sorry it was my fault, I forgot to import java before the mvn deploy :confused:</p>

**Comment by alastairpatrick** [comment]
<p>I had this problem too and it took me a while to figure out what “import java” means. This solved it. Add a setup-java step to the  job before the step that invokes maven:</p>
<pre><code class="lang-auto">-uses: actions/setup-java@v1
 with:
   java-version: 1.8

-name: Publish package
 run: mvn -B deploy
 env:
   GITHUB\_TOKEN: ${{ secrets.GITHUB_TOKEN }}
</code></pre>
